### ISSUE TITLE
`weights_init` usa `uniform(-1, 1)` e faz a rede convolucional divergir

### ISSUE THREAD

**Issue by luizth** [luizth/ppgca-dl-dqn#4]
## Onde

`src/network.py:11` e `src/network.py:15`

```python
torch.nn.init.uniform_(m.weight, a=-1.0, b=1.0)
```

## Causa

`uniform(-1, 1)` ignora o fan-in. Para camadas largas (`Linear(3136, 512)`) a variância acumulada explode.

## Evidência

Q-values na inicialização, mesma entrada:

```
DQN(conv) Q-values:    [ 16861.79  202514.31   9054.99  -95345.34 ]   (|max| = 2.03e+05)
default-init PyTorch:  [    -0.030    -0.029     -0.006     -0.035 ]
```

Cinco ordens de grandeza acima do razoável — o primeiro passo de gradiente diverge com qualquer learning rate útil. O MLP pequeno (`4 -> 128 -> 2`) ainda sobrevive, mas não é uma inicialização defensável.

## Nota adicional

`DQN.__init__` (`network.py:97`) chama `self.net.apply(weights_init)` **de novo** sobre a CNN e o MLP que já foram inicializados nos seus próprios construtores. Redundante.

## Correção

Usar a inicialização padrão do PyTorch, ou He/Kaiming (`init.kaiming_uniform_` com `nonlinearity="relu"`), que é o apropriado para as ReLUs usadas aqui.

**Comment by luizth** [comment]
Corrigido em 88c4f98 (branch `review-and-fix`).

`weights_init` passou a usar `init.kaiming_uniform_(m.weight, nonlinearity='relu')`, apropriado para as ReLUs da rede, e o `apply(weights_init)` redundante em `DQN.__init__` foi removido — a CNN e o MLP já se inicializam nos próprios construtores.

Efeito medido, mesma entrada e mesma seed:

```
Q-values na inicializacao:
  uniform(-1,1) antes : [154367.34  77755.40]
  He/Kaiming depois   : [     1.24      0.11]

25 passos de treino, lr=1e-4:
  uniform(-1,1) antes   loss=nan        |peso|max=nan     NaN/Inf=True
  He/Kaiming depois     loss=0.001809   |peso|max=0.1531  NaN/Inf=False
```

A rede não convergia devagar — ela saturava para NaN em menos de 25 passos. Regressão conferida no caminho MLP: CartPole, Acrobot, MountainCar e FrozenLake constroem com `|Q|` inicial entre 0.07 e 0.74 e treinam 120 passos sem NaN.
